### PR TITLE
Openshift upgrade 4 3

### PR DESCRIPTION
**Pull request by nathan-px**

### CODE REVIEW

File: content/portworx-install-with-kubernetes/openshift/operator/upgrade.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ aliases:
   - /portworx-install-with-kubernetes/on-premise/openshift/operator/upgrade/
 ---
 
+{{<info>}}
+**WARNING:** If you're upgrading OpenShift to 4.3, you must change Portworx before you can do so. See this[]() for more details.
```
Comment (piyush-nimbalkar) [inline content/portworx-install-with-kubernetes/openshift/operator/upgrade.md]: missing link to the new page

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+title: Preparing Portworx to upgrade to OpenShift 4.3 using the Operator
+keywords: portworx, container, kubernetes, storage, docker, k8s, pv, persistent disk, openshift
+description: Find out how to upgrade Portworx for OpenShift 4.3 using Operator.
+weight: 4
+---
+
+Before you can upgrade your OpenShift environment to 4.3, you must upgrade Portworx to 2.3.5 and expose a new port range so that Portworx can continue to operate.
+
+OpenShift 4.3 now reserves Portworx's previous port range of 9001 - 9099. To avoid a conflict with OpenShift 4.3, you must change the port range Portworx uses.
```
Comment (piyush-nimbalkar) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: *9001 - 9022

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+1. [Upgrade Portworx](/content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade/) to 2.3.5. 
+
+2. Open the port ranges on your cloud provider and operating system to allow traffic on ports 17000 - 17020.
+
+you must upgrade Portworx to version 2.3.5. and then open the Port ranges on your cloud provider and on your operating system to allow 17000 - 17020.
```
Comment (piyush-nimbalkar) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: remove this line

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+2. Change the port in the Portworx cluster spec in the operator to run on 17000 - 17020
```
Comment (piyush-nimbalkar) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: This is step 3.
"The ports should only be changed once all the Portworx nodes are upgraded to 2.3.5"

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+weight: 4
+---
+
+Before you can upgrade your OpenShift environment to 4.3, you must upgrade Portworx to 2.3.5 and expose a new port range so that Portworx can continue to operate.
+
+OpenShift 4.3 now reserves Portworx's previous port range of 9001 - 9099. To avoid a conflict with OpenShift 4.3, you must change the port range Portworx uses.
+
+1. [Upgrade Portworx](/content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade/) to 2.3.5. 
+
+2. Open the port ranges on your cloud provider and operating system to allow traffic on ports 17000 - 17020.
```
Comment (piyush-nimbalkar) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: *17001

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+you must upgrade Portworx to version 2.3.5. and then open the Port ranges on your cloud provider and on your operating system to allow 17000 - 17020.
+
+2. Change the port in the Portworx cluster spec in the operator to run on 17000 - 17020
+
+3. Once Portworx is running on 2.3.5, you may upgrade OpenShift to 4.3
```
Comment (piyush-nimbalkar) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: Once Portworx is upgraded to 2.3.5 and port range has shifted to 17001, you may upgrade ..

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md (hunk 4 above)
Comment (piyush-nimbalkar) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: Give steps to do this - 
```
3.a.
oc edit -n kube-system <storagecluster_name>

3.b.
apiVersion: core.libopenstorage.org/v1alpha1	      
kind: StorageCluster	      
metadata:	      
  name: portworx	        
  namespace: kube-system	        
spec:	      
  startPort: 17001
```

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+keywords: portworx, container, kubernetes, storage, docker, k8s, pv, persistent disk, openshift
+description: Find out how to upgrade Portworx for OpenShift 4.3 using Operator.
+weight: 4
+---
+
+Before you can upgrade your OpenShift environment to 4.3, you must upgrade Portworx to 2.3.5 and expose a new port range so that Portworx can continue to operate. Perform the following steps to upgrade Portworx to 2.3.5 in preparation for upgrading your OpenShift environment:
+
+1. Open the port ranges on your cloud provider and operating system to allow traffic on ports 17000 - 17020.
+
+2. [Upgrade Portworx](/content/portworx-install-with-kubernetes/openshift/operator/upgrade/#upgrade-portworx) to 2.3.5 and change the port in the same `StorageCluster` spec to run in the 17000 - 17020 range.
```
Comment (nathan-px) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: @piyush-nimbalkar Do you have the field name for the port range?

Comment (adityadani) [reply content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: Can we separate this into two separate steps?
Step 2: Upgrade Portworx to 2.3.5
Step 3: Change the port range from 900x to 17000x

**Comment by nathan-px** [comment]
@piyush-nimbalkar @adityadani take another look, I reworked it according to Piyush's feedback. It just needs the output for verification now.

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
+---
+title: Preparing Portworx to upgrade to OpenShift 4.3 using the Operator
+keywords: portworx, container, kubernetes, storage, docker, k8s, pv, persistent disk, openshift
+description: Find out how to upgrade Portworx for OpenShift 4.3 using Operator.
+weight: 4
+---
+
+Before you can upgrade your OpenShift environment to 4.3, you must upgrade Portworx to 2.3.5 and expose a new port range so that Portworx can continue to operate. Perform the following steps to upgrade Portworx to 2.3.5 in preparation for upgrading your OpenShift environment:
+
+2. [Upgrade Portworx](/content/portworx-install-with-kubernetes/openshift/operator/upgrade/#upgrade-portworx) to 2.3.5.
```
Comment (piyush-nimbalkar) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: This is step 1

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+2. Open the port ranges on your cloud provider and operating system to allow traffic on ports 17001 - 17020.
+
+3. Verify that all of your nodes have upgraded to 2.3.5:
```
Comment (piyush-nimbalkar) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: Lets swap step 2 and 3 - verify first and then open port ranges in cloud provider. It is more logical that way.

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+    ```text
+    oc get storagenode -n kube-system
+    ```
+    ```output
+
+    ```
+
+3. Change your Portworx cluster's start port:
```
Comment (piyush-nimbalkar) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: This is step 4

**Comment by piyush-nimbalkar** [comment]
The build seems to be failing due to broken links. I guess you don't need the "content/" prefix in urls.
Also add the output for 'oc get storaragenodes'

File: content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    ```output
+    NAME     ID                                     STATUS   VERSION           AGE
+    node-1   04904446-781b-4e12-b1ce-753a9e72dd49   Online   2.3.5.0-a5a0ba1   68s
+    node-2   b7a9e162-3166-475f-af6b-3860a4dc2b06   Online   2.3.5.0-a5a0ba1   2m
+    node-3   8c182888-280a-4738-abf8-3bccaadeb359   Online   2.3.5.0-a5a0ba1   3m
+    ```
+
+3. Open the port ranges on your cloud provider and operating system to allow traffic on ports 17001 - 17020.
+
+4. Change your Portworx cluster's start port:
```
Comment (vilasd) [inline content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: @piyush-nimbalkar do we need to mention them to be on operator:1.1.1 also ?

Comment (piyush-nimbalkar) [reply content/portworx-install-with-kubernetes/openshift/operator/openshift-upgrade.md]: Good catch. I forgot about the OCI mon readiness port fix that we added in 1.1.1.
@nathan-px Let's put the first step as upgrading the operator to 1.1.1 or above.
1. Upgrade the Portworx operator from OpenShift console to 1.1.1 or a newer version.

**Pull request merged**